src/main/java/seedu/duke/shooter/Shooter.java

```diff
@@ -0,0 +1,37 @@
+package seedu.duke.shooter;
+
+public class Shooter {
+    public String name;
+    private int power;
+    private int skill;
+
+    public Shooter(String name) {
+        this.name = name;
+    }
+    public void printSelfInfo(){
+        System.out.println("--------Player Info--------");
+        System.out.println("Name: "+this.name);
+        printPower();
+        printSkill();
+        System.out.println("---------------------------");
+    }
+
+    protected void printPower(){
+
+    }
+
+    protected void printSkill(){
+        System.out.print("Skill:");
+        for (int i=0; i<3;i++){
+            if (i<skill){
+                System.out.print(" 000");
+            }else{
+                System.out.print(" OOO");
+            }
+        }
+        System.out.println(" Level-Beginner");
+    }
+
+    public void printGoalBeforeShoot(){}
+    public void upgradePower(int level){}
```

Comment:
Again, why are these empty, and are they called at all?

Reply:
https://nus-cs2113-ay2324s2.github.io/website/schedule/week4/topics.html#W4-1
Here are the lecture slides for inheritance and polymorphism, please have a look before reviewing this part.